src/Bicep.Core.IntegrationTests/Scenarios/NullabilityTests.cs

```diff
@@ -193,18 +193,18 @@ private static IEnumerable<object[]> GetTemplatesWithSingleUnexpectedlyNullableV
 var arrayOfOneBuzz = [for i in first(arrayOfArrays)!: 'buzz']
 ",
                 LanguageConstants.Array,
-                TypeHelper.CreateTypeUnion(LanguageConstants.Null, new TypedArrayType(LanguageConstants.Int, default))
+                TypeHelper.CreateTypeUnion(LanguageConstants.Null, new TypedArrayType(TypeFactory.CreateIntegerType(0, 10), default))
             ),
 
             // nullably typed parameter defaults
             Case(
 @"
 param csv string
-param anotherParam string = first(split(csv, ','))
+param anotherParam string = first(filter(split(csv, ','), x => true))
 ",
 @"
 param csv string
-param anotherParam string = first(split(csv, ','))!
+param anotherParam string = first(filter(split(csv, ','), x => true))!
```

Comment:
Can we add tests for `last(split(` no warning behaviour?

Reply:
Added one to `FunctionResolverTests`:

![image](https://user-images.githubusercontent.com/705500/222268582-2849a473-6b40-4afd-8341-e2dfd3ff1a13.png)